File: study-1st/team2/Stack/Shinje/Tower.java
class Tower {
	public int[] solution(int[] heights) {
		int[] answer = new int[heights.length];
		Stack<Integer> stack = new Stack<Integer>();
		int len = heights.length;

		for (int i = 0; i < len; i++) {
			stack.push(heights[i]);
		}

		for (int i = len - 1; i >= 0; i--) {
			if (heights[i] > stack.peek()) {
				answer[i] = i + 1;
				stack.pop();
			} else {
				answer[i] = 0;
			}
		}

		return answer;
	}
}

File: study-1st/team2/DFS/Hyungsuk/tower.js
function solution(heights) {
    let heightsLength = heights.length;
    let answer = new Array(heightsLength).fill(0);
    for(let i = heightsLength - 1; i >= 0; i--) {
        for(let j = i - 1; j >= 0; j--) {
            if(heights[j] > heights[i]) {
                answer[i] = j + 1;
                break;
            }
        }      
    }
    return answer;
}
